crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -108,10 +105,7 @@ impl ProtocolFees {
             .into_iter()
             .collect::<Vec<_>>();
 
-        if self
-            .protocol_fee_exempt_addresses
-            .contains(&order.metadata.owner)
-        {
+        if surplus_capturing_jit_order_owners.contains(&order.metadata.owner.into()) {
```

Comment:
Do we actually want this line?

Previously, `protocol_fee_exempt_addresses` was used to exempt cow amm orders that were, at that time, part of an `Auction`.

Now, cow amm orders are not part of the `Auction`, therefore this line will not exclude any cow amm orders from protocol fees. On the contrary, if I am not wrong, it might skip attaching protocol fee to regular user orders that are posted with any cow amm order owner.

Reply:
CoW AMM addresses get passed as `surplus_capturing_jit_order_owners` [here](https://github.com/cowprotocol/services/pull/2799/files#diff-f23b91daf9f997f446211c1efa78147d5e2565c24913f08d3d4c07fc49e4ea87R256-R261).

Reply:
We pass them to drivers but they don't (shouldn't) have any effect on the Auction orders right?

Or better yet, do we want `surplus_capturing_jit_order_owners` to be exempt from protocol fee if they post an ordinary user order?

Reply:
Hmm, there are indeed 2 distinct concepts at play here:
1. being exempt from fees
2. allowing to capture surplus
And currently we use a single source of addresses for both. 🤔 
Overall I think cow amms should always be exempt from fees and should always be allowed to capture surplus regardless of whether they got indexed before the auction or got returned as a JIT order. So at the moment we don't have a correctness issue yet but as soon as we have addresses that capture surplus but still incur fees or vice versa we'd have a problem.

Reply:
> Or better yet, do we want surplus_capturing_jit_order_owners to be exempt from protocol fee if they post an ordinary user order?

This is the behavior we had. The idea was that all CoW AMM orders are counted for score and exempt from protocol fees.

Reply:
Right.

`whether they got indexed before the auction` - this doesn't exist anymore, right? If not, then that was the point of my comment, there is no need for the line ☝️ 

But I guess it doesn't hurt to keep it either.

Reply:
CoW AMM orders are permissionless and therefore it will always be possible for anyone to add them to our orderbook.